zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -261,6 +262,19 @@ def build_customprofile_field(customprofile_field: List[ZerverFieldsT], fields:
             customprofile_field.append(customprofilefield)
     return custom_field_map, customprofilefield_id
 
+def process_slack_custom_fields(user: ZerverFieldsT,
+                                slack_user_custom_field_map: ZerverFieldsT) -> None:
+    slack_user_custom_field_map[user['id']] = {}
+    if 'fields' in user['profile']:
+        # Make sure the content of fields is not 'None'
+        if user['profile']['fields']:
+            slack_user_custom_field_map[user['id']] = user['profile']['fields']
+
+    slack_custom_fields = ['phone', 'skype']
+    for field in slack_custom_fields:
+        if field in user['profile']:
+            slack_user_custom_field_map[user['id']][field] = {'value': user['profile'][field]}
+
```

Comment:
I would suggest you decompose this into first a single commit which extracts the body of the loop at line ~167 into this function, then adds the extra functionality in a subsequent commit.

To simplify the logic inside/outside the code, you might consider the following:
- instead of using "process" (ie. procedural) thinking, passing in the map, with a functional style you could return a single new map element in each call
- with this approach, you likely don't need the `user` itself inside the extracted method, just the profile dict?

Reply:
I think the current commit is fine and that refactoring is not required for this. However I am certainly open to refactoring this if it necessarily required.

Reply:
This is certainly reasonable as-is, it would just be cleaner, I think.

I'm happy to wait for another reviewer to weigh in on this.